### PR TITLE
Added tests for EndaomentAdmin

### PR DESCRIPTION
**Pull request by rheeger**
Tests are ready for EA.sol. 

Need to refactor down the road to incorporate getRoleAddress() and remove specific role getAddress() functions.

Love,
Robbie

### CODE REVIEW

File: contracts/endaomentAdmin.sol [hunk 1]
```diff
@@ -24,6 +24,14 @@ contract TwoStepOwnable {
     address indexed previousOwner,
     address indexed newOwner
   );
+ 
+  event TransferInitiated(
+    address indexed newOwner
+  );
+  
+  event TransferCancelled(
+    address indexed _newPotentialOwner
```
Comment (CantelopePeel) [inline contracts/endaomentAdmin.sol]: nit: no underscore

File: test/EndaomentAdmin.test.js [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+  expectEvent,
+  expectRevert,
+} = require("@openzeppelin/test-helpers");
+const { assert } = require("chai");
+
+const EndaomentAdmin = contract.fromArtifact("EndaomentAdmin");
+const OrgFactory = contract.fromArtifact("OrgFactory");
+const FundFactory = contract.fromArtifact("FundFactory");
+
+describe("EndaomentAmin.sol", function() {
```
Comment (CantelopePeel) [inline test/EndaomentAdmin.test.js]: typo EndaomentAdmin (no .sol)

File: test/EndaomentAdmin.test.js [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+
+  beforeEach(async function() {
+    this.endaomentAdmin = await EndaomentAdmin.new({ from: admin });
+  });
+
+  it("has defined contract address post-init", async function() {
+    assert.isDefined(this.endaomentAdmin.address);
+  });
+
+  it("set originator as the Owner on construct, allows retrieval of owner address", async function() {
```
Comment (CantelopePeel) [inline test/EndaomentAdmin.test.js]: simplify name

File: test/EndaomentAdmin.test.js [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+    const owner = await this.endaomentAdmin.getOwner({ from: admin });
+    assert.equal(owner, admin);
+  });
+
+  it("returns correct boolean for Owner based on provided address", async function() {
+    const isOwner = await this.endaomentAdmin.isOwner({ from: admin });
+    assert.isTrue(isOwner);
+    const isntOwner = await this.endaomentAdmin.isOwner({ from: newOwner });
```
Comment (CantelopePeel) [inline test/EndaomentAdmin.test.js]: s/isntOwner/isNotOwner/

File: test/EndaomentAdmin.test.js [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+    expectEvent(transferReceipt, "TransferInitiated", { newOwner });
+  });
+
+  it("denies non-Owner to intiate ownership transfer", async function() {
+    await expectRevert.unspecified(
+      this.endaomentAdmin.transferOwnership(newOwner, { from: newOwner })
+    );
+  });
+
+  it("prevents non-Owner from cancelling ownership transfer", async function() {
```
Comment (CantelopePeel) [inline test/EndaomentAdmin.test.js]: denies

File: test/EndaomentAdmin.test.js [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+    const cancelReceipt = await this.endaomentAdmin.cancelOwnershipTransfer({
+      from: admin,
+    });
+
+    expectEvent(cancelReceipt, "TransferCancelled", {
+      _newPotentialOwner: newOwner,
+    });
+  });
+
+  it("prevents non-Owner from cancelling ownership transfer", async function() {
```
Comment (CantelopePeel) [inline test/EndaomentAdmin.test.js]: ditto

File: test/EndaomentAdmin.test.js [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+      this.endaomentAdmin.cancelOwnershipTransfer({ from: newOwner })
+    );
+  });
+
+  it("allows Owner to cancel ownership transfer", async function() {
+    await this.endaomentAdmin.transferOwnership(newOwner, { from: admin });
+
+    const cancelReceipt = await this.endaomentAdmin.cancelOwnershipTransfer({
+      from: admin,
+    });
```
Comment (CantelopePeel) [inline test/EndaomentAdmin.test.js]: remove newlines around from: admin

File: test/EndaomentAdmin.test.js [hunk 8]
```diff
@@ -0,0 +1,265 @@
...
+  });
+
+  it("allows Owner to set Role.PAUSER, allows retrieval of PAUSER address", async function() {
+    await this.endaomentAdmin.setRole(1, pauser, { from: admin });
+    const pauserRole = await this.endaomentAdmin.getPauser({ from: admin });
+
+    assert.equal(pauserRole, pauser);
+  });
+
+  it("allows Owner to set Role.ACCOUNTANT, allows retrieval of ACCOUNTANT address", async function() {
```
Comment (CantelopePeel) [inline test/EndaomentAdmin.test.js]: replace with a single check of getRoleAddress for each of these with a test of a single role.

File: test/EndaomentAdmin.test.js [hunk 9]
```diff
@@ -0,0 +1,265 @@
...
+      from: admin,
+    });
+
+    assert.equal(orgFactoryRole, orgFactory.address);
+  });
+
+  it("allows Owner to remove Role", async function() {
+    await this.endaomentAdmin.setRole(0, admin, { from: admin });
+    const beforeRole = await this.endaomentAdmin.getAdmin();
+    await this.endaomentAdmin.removeRole(0, { from: admin });
```
Comment (CantelopePeel) [inline test/EndaomentAdmin.test.js:121]: Test that address is not set and that getRole fails

File: test/EndaomentAdmin.test.js [hunk 10]
```diff
@@ -0,0 +1,265 @@
...
+    await expectRevert.unspecified(
+      this.endaomentAdmin.removeRole(0, { from: newOwner })
+    );
+    const afterRole = await this.endaomentAdmin.getAdmin();
+
+    assert.equal(beforeRole, afterRole);
+  });
+
+  it("allows Owner to pause a given role", async function() {
+    await this.endaomentAdmin.setRole(1, pauser, { from: admin });
```
Comment (CantelopePeel) [inline test/EndaomentAdmin.test.js]: remove

File: test/EndaomentAdmin.test.js [hunk 11]
```diff
@@ -0,0 +1,265 @@
...
+  it("confirms if caller holds role ", async function() {
+    await this.endaomentAdmin.setRole(0, admin, { from: admin });
+    const holdsAdminRole = await this.endaomentAdmin.isRole(0, { from: admin });
+
+    assert.isTrue(holdsAdminRole);
+  });
+
+  it("confirms if caller does not hold role ", async function() {
+    await this.endaomentAdmin.setRole(0, admin, { from: admin });
+    const holdsAdminRole = await this.endaomentAdmin.isRole(0, {
```
Comment (CantelopePeel) [inline test/EndaomentAdmin.test.js]: from bit on same line

File: test/Fund.test.js [hunk 12]
```diff
@@ -12,7 +12,7 @@ const FundFactory = contract.fromArtifact("FundFactory");
 const OrgFactory = contract.fromArtifact("OrgFactory");
 const Fund = contract.fromArtifact("Fund");
 
-describe("Fund", function() {
+describe("Fund.sol", function() {
```
Comment (CantelopePeel) [inline test/Fund.test.js]: ditto

File: test/OrgFactory.test.js [hunk 13]
```diff
@@ -11,7 +11,7 @@ const EndaomentAdmin = contract.fromArtifact("EndaomentAdmin");
 const OrgFactory = contract.fromArtifact("OrgFactory");
 const Org = contract.fromArtifact("Org");
 
-describe("OrgFactory", function() {
+describe("OrgFactory.sol", function() {
```
Comment (CantelopePeel) [inline test/OrgFactory.test.js]: ditto

File: test/fundFactory.test.js [hunk 14]
```diff
@@ -11,7 +11,7 @@ const EndaomentAdmin = contract.fromArtifact("EndaomentAdmin");
 const FundFactory = contract.fromArtifact("FundFactory");
 const Fund = contract.fromArtifact("Fund");
 
-describe("FundFactory", function() {
+describe("FundFactory.sol", function() {
```
Comment (CantelopePeel) [inline test/fundFactory.test.js]: ditto

**Review by CantelopePeel — CHANGES_REQUESTED** [review]

File: test/EndaomentAdmin.test.js (hunk 3 above)
Comment (rheeger) [reply test/EndaomentAdmin.test.js]: which name?

Comment (CantelopePeel) [reply test/EndaomentAdmin.test.js]: set originator as the Owner on construct, allows retrieval of owner address

File: contracts/Fund.sol [hunk 15]
```diff
@@ -78,9 +78,9 @@ contract Fund is Administratable {
         * @param  orgFactoryContractAddress Address of the OrgFactory contract.
         */
         function checkRecipient(address recipient, address orgFactoryContractAddress) public view returns (bool) {
-            OrgFactory x = OrgFactory ( orgFactoryContractAddress );
+            OrgFactory orgFactory = OrgFactory ( orgFactoryContractAddress );
```
Comment (CantelopePeel) [inline contracts/Fund.sol]: Remove spaces around parens

File: contracts/endaomentAdmin.sol [hunk 16]
```diff
@@ -18,12 +18,20 @@ import "./interfaces/IEndaomentAdmin.sol";
  */
 contract TwoStepOwnable {
   address private _owner;
-  address private _newPotentialOwner;
+  address private newPotentialOwner;
```
Comment (CantelopePeel) [inline contracts/endaomentAdmin.sol]: this should still be underscored: _newPotentialOwner;

File: contracts/endaomentAdmin.sol [hunk 17]
```diff
@@ -18,12 +18,20 @@ import "./interfaces/IEndaomentAdmin.sol";
...
     address indexed previousOwner,
     address indexed newOwner
   );
+ 
+  event TransferInitiated(
+    address indexed newOwner
+  );
+  
+  event TransferCancelled(
+    address indexed newPotentialOwner
```
Comment (CantelopePeel) [inline contracts/endaomentAdmin.sol:33]: but not underscored here

**Review by CantelopePeel — CHANGES_REQUESTED** [review]

File: test/EndaomentAdmin.test.js (hunk 3 above)
Comment (rheeger) [reply test/EndaomentAdmin.test.js]: like you think this could be worded better basically? yea?

**Review by CantelopePeel — APPROVED** [review]

**Pull request merged**